src/expression.d

```diff
@@ -2588,16 +2588,27 @@ public:
         return this;
     }
 
+    /****************************************
+     * Check that the expression has a valid type.
+     * Returns:
+     *      true if the expression is not valid.
```

Comment:
Added a comment line to mention error generation.

Currently, all functions `bool checkSomething()` in dmd will test a condition "Something". And it will return `false` if the test result is no problem. Otherwise, report a relevant error and returns `true`. It's a normal idiom in dmd.

And, as I said in the summary, `checkType` function is strongly related with `checkValue`. If we choose other name for that, we should also rename `checkValue` function.